specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,542 @@
+            # be folded
+            source_bin_i = spec[i]
+
+            # get the current number of groups
+            current_num_groups = self.n_grp[i]
+
+            # loop over the current number of groups
+            for j in range(current_num_groups):
+
+                current_num_chans = int(self.n_chan[k])
+
+                if current_num_chans == 0:
+                    k += 1
+                    resp_idx += current_num_chans
+                    continue
+
+
+                else:
+                    # get the right index for the start of the counts array
+                    # to put the data into
+                    counts_idx = int(self.f_chan[k] - self.offset)
+                    # this is the current number of channels to use
+
+                    k += 1
+                    # add the flux to the subarray of the counts array that starts with
+                    # counts_idx and runs over current_num_chans channels
+                    counts[counts_idx:counts_idx +
+                                      current_num_chans] += self.matrix[resp_idx:resp_idx +
+                                                                                 current_num_chans] * \
+                                                                np.float(source_bin_i)
+                    # iterate the response index for next round
+                    resp_idx += current_num_chans
+
+
+        return counts[:self.detchans]
+
+
+class ARF(object):
+
+    def __init__(self, filename):
+
+        self._load_arf(filename)
+        pass
+
+    def _load_arf(self, filename):
+        """
+        Load an ARF from a FITS file.
+
+        Parameters
+        ----------
+        filename : str
+            The file name with the RMF file
+
+        Attributes
+        ----------
+        filename : str
+            The file name that the ARF was drawn from
+
+        e_low : numpy.ndarray
+            The lower edges of the energy bins
+
+        e_high : numpy.ndarray
+            The upper edges of the energy bins
+
+        e_unit : astropy.units.Unit
+            Description of the energy units used
+
+        specresp : numpy.ndarray
+            Description of the energy dependent telescope response area
+
+        fracexpo : float or numpy.ndarray
+            Fractional exposure time for the spectrum
+            (sometimes constant, sometimes dependent on spectral channel).
+            These values are stored for reference; generally, they are already
+            accounted for in the specresp array.
+
+        exposure :
+            Average exposure time for the dataset
+            (takes telescope dithering into account)
+        """
+        # open the FITS file and extract the MATRIX extension
+        # which contains the redistribution matrix and
+        # anxillary information
+        hdulist = fits.open(filename)
+        self.filename = filename
+
+        h = hdulist["SPECRESP"]
```

Comment:
Have 'SPECRESP' has a keyword option

Reply:
Pushing back on this gently because it is so standard. I can't think of a reason a user would ever want to change this value. It is so standard, we named the attribute after it.

Pinging @dhuppenkothen for comments, since this portion is widely her doing.

Reply:
It's been a while since I looked at this, but I have the vague memory that there might be instruments that don't call this `SPECRESP`, in which case having it as a keyword might be helpful?

The code I wrote was never meant to be fully inclusive of every instruments; it was a prototype I wanted to get work on a single type of data, and the possibly extend it to more types of data. As such, don't assume that I've necessarily picked the most general way of implementing things everywhere. :)

Reply:
Ahh, okay. I thought we had tried it on enough telescopes. I will add the keyword and change the attribute name to something informative, like `effective_area`